realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -233,54 +363,125 @@ public void copyFromIsNotLive() {
     }
 
     /**
-     * Assure that an attempt to change the value of a managed RealmInteger, outside a transaction, fails.
+     * Assure that an attempt to assign a RealmInteger field, outside a transaction, fails.
      */
-    @Ignore("not yet implemented")
     @Test
-    public void updateOutsideTransactionThrows() {
+    public void assignOutsideTransactionThrows() {
         realm.beginTransaction();
-        realm.createObject(AllTypes.class).setColumnRealmInteger(new RealmInteger(5));
+        Counters managedCounters = realm.createObject(Counters.class);
+        managedCounters.setColumnRealmInteger(RealmInteger.valueOf(5));
         realm.commitTransaction();
 
-        RealmInteger managedRI = realm.where(AllTypes.class).findFirst().getColumnRealmInteger();
         try {
-            managedRI.set(1);
-            fail("Setting a managed RealmInteger outside a transaction should fail");
+            managedCounters.setColumnRealmInteger(RealmInteger.valueOf(7));
+            fail("Assigning a managed RealmInteger outside a transaction should fail");
         } catch (IllegalStateException e) {
             assertTrue(e.getMessage().contains("must be in a transaction"));
         }
+    }
+
+    /**
+     * Semantics definition from #4266.
+     */
+    @Test
+    @SuppressWarnings("ReferenceEquality")
+    public void unmanaged_semantics() {
+        Counters obj1 = new Counters();
+        obj1.columnRealmInteger = RealmInteger.valueOf(0);
+
+        Counters obj2 = new Counters();
+        obj2.columnRealmInteger = RealmInteger.valueOf(30);
+
+        obj1.columnRealmInteger.increment(1);
+
+        obj2.columnRealmInteger = obj1.columnRealmInteger;
+
+        assertEquals(1, obj2.columnRealmInteger);
+        assertTrue(obj1.columnRealmInteger.equals(obj2.columnRealmInteger));
+        assertTrue(obj1.columnRealmInteger == obj2.columnRealmInteger);
+    }
+
+    /**
+     * Semantics definition from #4266.
+     */
+    @Test
+    @SuppressWarnings("ReferenceEquality")
+    public void mixed_semantics() {
+        realm.beginTransaction();
+        Counters obj1 = realm.createObject(Counters.class);
+        realm.commitTransaction();
+
+        Counters obj2 = new Counters();
+        obj1.columnRealmInteger = RealmInteger.valueOf(30);
```

Comment:
This should crash? It should never be allowed to modify a managed object outside a transaction